### ISSUE TITLE
avc denials

### ISSUE THREAD

**Issue by fuqianz** [net-snmp/net-snmp#325]
avc: denied { module_request } for pid=39960 comm="snmpd" kmod="netdev-usb0" 
  scontext=system_u:system_r:snmpd_t:s0 tcontext=system_u:system_r:kernel_t:s0 tclass=system permissive=0

Any idea on why this happened?

Thanks

**Comment by bvanassche** [comment]
Please ask help from a CentOS SELinux expert. I think this means that the CentOS SELinux policies need to be improved.

**Comment by sargenthp** [comment]
Does net-snmp maintain the SELinux rules, or is that something the distros do?  I am having similar issues with hrPartition.

**Comment by hardaker** [comment]
No, we don't maintain selinux rules and can't really.  We could for our base installation, but most repos have their own set of configure options that place files in different locations, etc.  It would be reasonable to build selinux policy files and install them I suppose, but no one has ever gone through the effort of doing that.

**Comment by sargenthp** [comment]
OK.  Thank you.  I will leave this here, just incase others run into the same issue.  I was having issues with RHEL 8.7 and SELinux was blocking the request.  I do have a ticket with RHEL, so hopefully they will update the rules for RHEL, and there for CentOS, and Rocky.  But not sure if they will do anything for version 7.

Create the file snmpd_blk_file.te with the following contents....
```
module snmpd_blk_file 1.0;

require {
	type snmpd_t;
	type fixed_disk_device_t;
	type removable_device_t;
	class blk_file { getattr ioctl open read };
}

#============= snmpd_t ==============
allow snmpd_t fixed_disk_device_t:blk_file { getattr ioctl open read };
allow snmpd_t removable_device_t:blk_file { getattr ioctl open read };
```

Then need to build the files to import into selinux...
```
# checkmodule -M -m -o snmpd_blk_file.mod snmpd_blk_file.te
# semodule_package -o snmpd_blk_file.pp -m snmpd_blk_file.mod
```

Then import the rules...
`# semodule -i snmpd_blk_file.pp`
